crates/wac-graph/src/plug.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{types::SubtypeChecker, CompositionGraph, PackageId};
+use thiserror::Error;
+
+/// Represents an error that can occur when plugging components together.
+#[derive(Debug, Error)]
+pub enum PlugError {
+    /// The socket component had no matching imports for the plugs that were provided
+    #[error("the socket component had no matching imports for the plugs that were provided")]
+    NoPlugHappened,
+    /// An error occurred when building the composition graph
+    #[error("an error occurred when building the composition graph")]
+    GraphError {
+        /// The underlying graph error
+        #[source]
+        source: anyhow::Error,
+    },
+}
+
+/// Take the exports of the plug components and plug them into the socket component.
```

Comment:
Can you please document that the `PackageId`s in `plugs` and `socket` are expected to be registered with the composition graph already?